### PR TITLE
Localhost filter includes url

### LINKED ISSUE #4800: Localhost filter should consider affected URL
Right now the "localhost" filter only consider's the affected User's IP: https://github.com/getsentry/sentry/blob/master/src/sentry/filters/localhost.py

But users are also expecting that this should filter server errors triggered from a server running on localhost (e.g. local development).

See also: #4729, #4762

### PR DESCRIPTION
**Pull request by MeredithAnya**
Basically copypasta from #4762 
Fixes #4800 

@mattrobenolt

### CODE REVIEW

File: src/sentry/filters/localhost.py [hunk 1]
```diff
@@ -16,5 +19,20 @@ def get_ip_address(self, data):
...
+        try:
+            http = data['sentry.interfaces.Http']
+            if http and http['url']:
+                return http['url']
+            return ''
+        except KeyError:
+            return ''
+
+    def get_domain(self, data):
+        matchObj = DOMAIN_FROM_URL.match(self.get_url(data))
```
Comment (mattrobenolt) [inline src/sentry/filters/localhost.py]: I'd rather we not use a regular expression here, and instead use `urlparse`

File: src/sentry/filters/localhost.py [hunk 2]
```diff
@@ -16,5 +19,20 @@ def get_ip_address(self, data):
         except KeyError:
             return ''
 
+    def get_url(self, data):
+        try:
+            http = data['sentry.interfaces.Http']
```
Comment (mattrobenolt) [inline src/sentry/filters/localhost.py]: I'd simplify this to just `return data['sentry.interfaces.Http']['url']`

File: tests/sentry/filters/test_localhost.py [hunk 3]
```diff
@@ -31,3 +34,20 @@ def test_does_not_filter_external_ip(self):
...
         assert not self.apply_filter({})
+
+    def test_filters_localhost_domain(self):
+        data = self.get_mock_data(url='http://localhost/something.html')
+        assert self.apply_filter(data)
+
+        data = self.get_mock_data(url='https://localhost')
+        assert self.apply_filter(data)
+
+        data = self.get_mock_data(url='http://localhost?domain_in_args=example.com')
```
Comment (mattrobenolt) [inline tests/sentry/filters/test_localhost.py]: This test is bad since it's not a valid url.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
We need to also have tests that cover the 127.0.0.1 case, as well as tests that cover the missing key path.

I also want to re-iterate that this will never solve the actual problem and this is still quite hacky solution. Granted, there isn't much better we can do. But we should communicate this in the product so set expectations.

**Comment by MeredithAnya** [comment]
this probs needs more tests but wanted to address the other changes since its been some time

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**